cli/cliui/select.go

```diff
@@ -75,33 +75,172 @@ func Select(inv *serpent.Invocation, opts SelectOptions) (string, error) {
 		return opts.Options[0], nil
 	}
 
-	var defaultOption interface{}
-	if opts.Default != "" {
-		defaultOption = opts.Default
+	initialModel := selectModel{
+		search:     textinput.New(),
+		hideSearch: opts.HideSearch,
+		options:    opts.Options,
+		height:     opts.Size,
+		message:    opts.Message,
 	}
 
+	if initialModel.height == 0 {
+		initialModel.height = 7
+	}
+
+	initialModel.search.Prompt = ""
+	initialModel.search.Focus()
+
+	m, err := tea.NewProgram(
+		initialModel,
+		tea.WithInput(inv.Stdin),
+		tea.WithOutput(inv.Stdout),
+	).Run()
+
 	var value string
-	err := survey.AskOne(&survey.Select{
-		Options:  opts.Options,
-		Default:  defaultOption,
-		PageSize: opts.Size,
-		Message:  opts.Message,
-	}, &value, survey.WithIcons(func(is *survey.IconSet) {
-		is.Help.Text = "Type to search"
-		if opts.HideSearch {
-			is.Help.Text = ""
-		}
-	}), survey.WithStdio(fileReadWriter{
-		Reader: inv.Stdin,
-	}, fileReadWriter{
-		Writer: inv.Stdout,
-	}, inv.Stdout))
-	if errors.Is(err, terminal.InterruptErr) {
-		return value, Canceled
+	if m, ok := m.(selectModel); ok {
+		if m.canceled {
+			return value, Canceled
+		}
+		value = m.selected
```

Comment:
Nit: This could be simplified to `return m.selected, nil` instead of going through an intermediary variable.

Reply:
Good spot, yes I've done this now